Bio/PDB/PDBList.py

```diff
@@ -271,16 +269,16 @@ def retrieve_pdb_file(
 
         # Get the compressed PDB structure
         pdb_code = pdb_code.lower()
-        archive = {
+        archive_dict = {
             "pdb": f"pdb{pdb_code}.ent.gz",
             "mmCif": f"{pdb_code}.cif.gz",
             "xml": f"{pdb_code}.xml.gz",
             "mmtf": f"{pdb_code}",
             "bundle": f"{pdb_code}-pdb-bundle.tar.gz",
         }
-        archive_fn = archive[file_format]
+        archive = archive_dict[file_format]
 
-        if file_format not in archive.keys():
+        if file_format not in archive_dict.keys():
```

Comment:
Isn't this never going to happen since line 29 will trigger a KeyError?

Reply:
Also style wise, change ``if file_format not in archive_dict.keys():`` to ``if file_format not in archive_dict:`` and likewise anywhere else if you want to modernize these bits.

Reply:
Good catch, I will remove it.

And thanks for the style recommendation.

Reply:
Thank you. This is improved in the next set of changes, and I manually tested it.

Input
```python
from Bio.PDB import PDBList

pdbl = PDBList()
pdbl.retrieve_pdb_file("test", file_format="test")
```

Output
```
Traceback (most recent call last):
  File "/Users/willtyler/Applications/PyCharm Community Edition.app/Contents/plugins/python-ce/helpers/pydev/pydevconsole.py", line 364, in runcode
    coro = func()
           ^^^^^^
  File "<input>", line 4, in <module>
  File "/Users/willtyler/Desktop/biopython/Bio/PDB/PDBList.py", line 281, in retrieve_pdb_file
    raise ValueError(
ValueError: Specified file_format test does not exist or is not supported. Please use one of the following: pdb, mmCif, xml, mmtf, bundle.
```